standard/wasi-filesystem/witx/typenames.witx

```diff
@@ -0,0 +1,708 @@
+;; Type names used by low-level WASI interfaces.
+;;
+;; Some content here is derived from [CloudABI](https://github.com/NuxiNL/cloudabi).
+;;
+;; This is a `witx` file. See [here](https://github.com/WebAssembly/WASI/tree/master/docs/witx.md)
+;; for an explanation of what that means.
+
+;;; An array size.
+;;;
+;;; Note: This is similar to `size_t` in POSIX.
+(typename $size (@witx usize))
+
+;;; Non-negative file size or length of a region within a file.
+(typename $filesize u64)
+
+;;; Timestamp in nanoseconds.
+(typename $timestamp u64)
+
+;;; Identifiers for clocks.
+(typename $clockid
+  (enum (@witx tag u32)
+    ;;; The clock measuring real time. Time value zero corresponds with
+    ;;; 1970-01-01T00:00:00Z.
+    $realtime
+    ;;; The store-wide monotonic clock, which is defined as a clock measuring
+    ;;; real time, whose value cannot be adjusted and which cannot have negative
+    ;;; clock jumps. The epoch of this clock is undefined. The absolute time
+    ;;; value of this clock therefore has no meaning.
+    $monotonic
+  )
+)
+
+;;; Error codes returned by functions.
+;;; Not all of these error codes are returned by the functions provided by this
+;;; API; some are used in higher-level library layers, and others are provided
+;;; merely for alignment with POSIX.
+(typename $errno
+  (enum (@witx tag u16)
+    ;;; No error occurred. System call completed successfully.
+    $success
+    ;;; Argument list too long. This is similar to `E2BIG` in POSIX.
+    $too_big
+    ;;; Permission denied.
+    $access
+    ;;; Address in use.
+    $addrinuse
```

Comment:
As too_big are used, may `addrinuse` -> `addr_in_use`?
And also others

Reply:
As above, I've now renamed `too_big` to `toobig`.

Reply:
> As above, I've now renamed `too_big` to `toobig`.

LGTM